### PR TITLE
doc: comment beta status

### LINKED ISSUE #1033: Update readme
https://github.com/istio/ztunnel/blob/master/README.md:

> Ztunnel provides an experimental implementation of the ztunnel component of [ambient mesh](https://istio.io/latest/blog/2022/introducing-ambient-mesh/).
>
> Note: `istio/ztunnel` is currently intended for experimental usage only.

etc

### PR DESCRIPTION
**Pull request by daixiang0**
Fix #1033

### CODE REVIEW

**Comment by kfaseela** [comment]
@craigbox okay to merge this now? Or we need to wait?

**Comment by craigbox** [comment]
Pass, I defer to a ztunnel maintainer

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
